snuba/attribution/log.py

```diff
@@ -33,3 +49,42 @@ def log_attribution(attr_data: AttributionData) -> None:
             "table": q.table,
         }
         metrics.increment("log", q.bytes_scanned, tags=tags)
+
+
+def _record_attribution_delivery_callback(
+    error: Optional[KafkaError], message: KafkaMessage
+) -> None:
+    metrics.increment(
+        "record_attribution.delivery_callback",
+        tags={"status": "success" if error is None else "failure"},
+    )
+
+    if error is not None:
+        logger.warning("Could not record attribution due to error: %r", error)
+
+
+def record_attribution(attr_data: AttributionData) -> None:
+    global kfk
+
+    log_attribution_to_metrics(attr_data)
+
+    try:
+        data = asdict(attr_data)
+        data[
+            "app_id"
+        ] = attr_data.app_id.key  # Don't record the entire AppID, just the key
+        data_str = safe_dumps(data)
+
+        if kfk is None:
+            kfk = Producer(build_default_kafka_producer_configuration())
+
+        kfk.poll(0)  # trigger queued delivery callbacks
+        kfk.produce(
+            settings.KAFKA_TOPIC_MAP.get(
+                Topic.ATTRIBUTION.value, Topic.ATTRIBUTION.value
+            ),
+            data_str.encode("utf-8"),
+            on_delivery=_record_attribution_delivery_callback,
+        )
+    except Exception as ex:
+        logger.exception("Could not record attribution due to error: %r", ex)
```

Comment:
I would be more comfortable by setting this to logger.warning. If we don't log attribution for some reason, we don't want to get alerted/paged about it.